diff --git a/src/architecture.cpp b/src/architecture.cpp
index eca306f..2162c19 100644
--- a/src/architecture.cpp
+++ b/src/architecture.cpp
@@ -1922,7 +1922,7 @@ gfx9_base_t::register_address (const cwsr_descriptor_t &descriptor,
       || ((regnum == amdgpu_regnum_t::EXEC_64
            || regnum == amdgpu_regnum_t::XNACK_MASK_64)
           && lane_count != 64))
-    return {};
+    return std::nullopt;
 
   /* Rename registers that map to the hwreg block.  */
   switch (regnum)
@@ -1978,7 +1978,7 @@ gfx9_base_t::register_address (const cwsr_descriptor_t &descriptor,
       && regnum <= amdgpu_regnum_t::LAST_SGPR
       && (regnum - amdgpu_regnum_t::S0)
              >= (std::min (108ul, sgpr_count) - scalar_alias_count ()))
-    return {};
+    return std::nullopt;
 
   /* Rename registers that alias to sgprs.  */
   if ((lane_count == 32 && regnum == amdgpu_regnum_t::VCC_32)
@@ -2050,7 +2050,7 @@ gfx9_base_t::register_address (const cwsr_descriptor_t &descriptor,
       return vgprs_addr + (regnum - amdgpu_regnum_t::V0_64) * vgpr_size;
     }
 
-  return {};
+  return std::nullopt;
 }
