diff --git a/lib/comgr/CMakeLists.txt b/lib/comgr/CMakeLists.txt
index f31bdeb..8d5a681 100644
--- a/lib/comgr/CMakeLists.txt
+++ b/lib/comgr/CMakeLists.txt
@@ -342,7 +342,7 @@ if(CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR)
 
   # Debian package specific variables
   set(CPACK_DEBIAN_PACKAGE_HOMEPAGE "https://github.com/RadeonOpenCompute/ROCm-CompilerSupport")
-  set(CPACK_DEBIAN_PACKAGE_DEPENDS "libtinfo5, rocm-core")
+  set(CPACK_DEBIAN_PACKAGE_DEPENDS "libtinfo6, rocm-core")
   if (DEFINED ENV{CPACK_DEBIAN_PACKAGE_RELEASE})
     set(CPACK_DEBIAN_PACKAGE_RELEASE $ENV{CPACK_DEBIAN_PACKAGE_RELEASE})
   else()
