diff --git a/include/hip/amd_detail/amd_hip_runtime.h b/include/hip/amd_detail/amd_hip_runtime.h
index 41cdf24e..eb86fa63 100644
--- a/include/hip/amd_detail/amd_hip_runtime.h
+++ b/include/hip/amd_detail/amd_hip_runtime.h
@@ -451,7 +451,7 @@ hc_get_workitem_absolute_id(int dim)
 // path instead of the clang include path itself.
 #include <include/cuda_wrappers/algorithm>
 #include <include/cuda_wrappers/complex>
-#include <include/cuda_wrappers/new>
+//#include <include/cuda_wrappers/new>
 #undef __CUDA__
 #pragma pop_macro("__CUDA__")
 #endif // !_OPENMP || __HIP_ENABLE_CUDA_WRAPPER_FOR_OPENMP__
