diff --git a/library/src/blas_ex/rocblas_gemm_ext2.hpp b/library/src/blas_ex/rocblas_gemm_ext2.hpp
index ffed55d3..7dbb2282 100644
--- a/library/src/blas_ex/rocblas_gemm_ext2.hpp
+++ b/library/src/blas_ex/rocblas_gemm_ext2.hpp
@@ -4,13 +4,13 @@
 #ifndef __ROCBLAS_GEMM_EXT2_HPP
 #define __ROCBLAS_GEMM_EXT2_HPP
 
+#include "rocblas_gemm_ex.hpp"
 // This functionality is only availble when using the new Tensile client
 #ifdef USE_TENSILE_HOST
 
 #include "handle.hpp"
 #include "logging.hpp"
 #include "rocblas.h"
-#include "rocblas_gemm_ex.hpp"
 #include "utility.hpp"
 
 template <typename Ti, typename To, typename Tc>
