diff --git a/rocprim/include/rocprim/device/device_histogram.hpp b/rocprim/include/rocprim/device/device_histogram.hpp
index f1b6a0b..398ac6f 100644
--- a/rocprim/include/rocprim/device/device_histogram.hpp
+++ b/rocprim/include/rocprim/device/device_histogram.hpp
@@ -24,6 +24,7 @@
 #include <cmath>
 #include <type_traits>
 #include <iterator>
+#include <thread>
 
 #include "../config.hpp"
 #include "../functional.hpp"
diff --git a/rocprim/include/rocprim/device/device_transform.hpp b/rocprim/include/rocprim/device/device_transform.hpp
index 12cb213..b561436 100644
--- a/rocprim/include/rocprim/device/device_transform.hpp
+++ b/rocprim/include/rocprim/device/device_transform.hpp
@@ -24,6 +24,7 @@
 #include <algorithm>
 #include <type_traits>
 #include <iterator>
+#include <thread>
 
 #include "../config.hpp"
 #include "../detail/various.hpp"
