diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4199ef5..cff8de6 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -34,7 +34,7 @@ set(SHARE_INSTALL_PREFIX
 find_program (GIT NAMES git)
 
 ## Setup the package version based on git tags.
-set(PKG_VERSION_GIT_TAG_PREFIX "rsmi_pkg_ver")
+set(PKG_VERSION_GIT_TAG_PREFIX "rocm")
 get_package_version_number("5.0.0" ${PKG_VERSION_GIT_TAG_PREFIX} GIT)
 message("Package version: ${PKG_VERSION_STR}")
 set(${AMD_SMI_LIBS_TARGET}_VERSION_MAJOR "${VERSION_MAJOR}")
