diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8bdc698..f0d73ef 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -23,7 +23,7 @@ set(ROCM_SMI_LIB_NAME "lib${ROCM_SMI_TARGET}")
 find_program (GIT NAMES git)
 
 ## Setup the package version based on git tags.
-set(PKG_VERSION_GIT_TAG_PREFIX "rsmi_pkg_ver")
+set(PKG_VERSION_GIT_TAG_PREFIX "rocm")
 get_package_version_number("1.0.0" ${PKG_VERSION_GIT_TAG_PREFIX} GIT)
 message("Package version: ${PKG_VERSION_STR}")
 set(${AMD_SMI_LIBS_TARGET}_VERSION_MAJOR "${VERSION_MAJOR}")
